sync.py

```diff
@@ -129,10 +129,16 @@ def _check_dir_for_changes(self, dir_path):
 
     def poll_for_changes(self, wait_time=1):
         while True:
-            self.pinger.ping()
-            self._check_dir_for_changes(self.base_path)
-            self._first_pass = False
-            time.sleep(wait_time)  # Poll for changes every `wait_time` seconds
+            try:
+                self.pinger.ping()
+                self._check_dir_for_changes(self.base_path)
+                self._first_pass = False
+            except requests.exceptions.ConnectionError:
+                print(f"Failed\nCould not reach {SERVER_URL}")
+            except Exception:
+                print("Unknown error")
```

Comment:
Will this catch `KeyboardInterrupt` too, and make the script hard to terminate?

Reply:
No, `KeyboardInterrupt` is at the same level as `Exception`, they're both subclasses of `BaseException`

Reply:
https://docs.python.org/3/library/exceptions.html#exception-hierarchy